broker/lib/jobs/BnRStatusPollerJob.js

```diff
@@ -126,7 +136,117 @@ class BnRStatusPollerJob extends BaseJob {
+          return Promise
+            .try(() => eventmesh
+              .server
+              .updateOperationResult({
+                operationName: 'backup',
+                operationType: 'defaultbackups',
+                operationId: instanceInfo.backup_guid,
+                value: operationStatusResponse
+              }))
+            .then(() => bosh.director.getDirectorConfig(instanceInfo.deployment))
+            .then(directorConfig => {
+              const lock_deployment_max_duration = directorConfig.lock_deployment_max_duration;
+              if (backup_triggered_duration > lock_deployment_max_duration) {
+                //Operation timed out
+                if (!instanceInfo.abortStartTime) {
+                  //Operation not aborted. Aborting operation and with abort start time
+                  // re-registering statupoller job
+                  let abortStartTime = new Date().toISOString();
+                  instanceInfo.abortStartTime = abortStartTime;
+                  return DirectorManager
+                    .load(plan)
+                    .then(directorManager => directorManager.abortLastBackup(instanceInfo, true))
+                    .then(() => DirectorManager.registerBnRStatusPoller(job_data, instanceInfo))
+                    .then(() => {
+                      operationStatusResponse.state = CONST.OPERATION.ABORTING;
+                      return operationStatusResponse;
+                    });
+                } else {
+                  // Operation aborted
+                  const currentTime = new Date();
+                  const abortDuration = (currentTime - new Date(instanceInfo.abortStartTime));
+                  if (abortDuration < config.backup.abort_time_out) {
+                    logger.info(`${operationName} abort is still in progress on : ${deployment} for guid : ${backup_guid}`);
+                    operationStatusResponse.state = CONST.OPERATION.ABORTING;
+                  } else {
+                    operationStatusResponse.state = CONST.OPERATION.ABORTED;
+                    logger.info(`Abort ${operationName} timed out on : ${deployment} for guid : ${backup_guid}. Flagging ${operationName} operation as complete`);
+                    operationStatusResponse.operationTimedOut = true;
+                    operationStatusResponse.operationFinished = true;
+                  }
+                  return operationStatusResponse;
+                }
+              } else {
+                // Backup not timedout and still in-porogress
+                return operationStatusResponse;
+              }
+            });
+        }
+      })
+      .tap(operationStatusResponse => {
+        return operationStatusResponse.operationFinished ?
+          this.doPostFinishOperation(operationStatusResponse, operationName, instanceInfo) :
+          Promise.resolve(operationStatusResponse);
```

Comment:
Unnecessary `Promise.resolve`

Reply:
Removed

Reply:
Minor comment: The `Promise.resolve` seems to be there still